solidity/contracts/l2/L2TBTC.sol

```diff
@@ -0,0 +1,242 @@
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+// TODO: Proper documentation (contract, fields, @params)
+contract L2TBTC is
+    ERC20Upgradeable,
+    ERC20BurnableUpgradeable,
+    ERC20PermitUpgradeable,
+    OwnableUpgradeable,
+    PausableUpgradeable
+{
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    /// @notice Indicates if the given address is a minter. Only minters can
+    ///         mint the token.
+    mapping(address => bool) public isMinter;
+
+    /// @notice List of all minters.
+    address[] public minters;
+
+    /// @notice Indicates if the given address is a guardian. Only guardians can
+    ///         pause token mints and burns.
+    mapping(address => bool) public isGuardian;
+
+    /// @notice List of all guardians.
+    address[] public guardians;
+
+    event MinterAdded(address indexed minter);
+    event MinterRemoved(address indexed minter);
+
+    event GuardianAdded(address indexed guardian);
+    event GuardianRemoved(address indexed guardian);
+
+    modifier onlyMinter() {
+        require(isMinter[msg.sender], "Caller is not a minter");
+        _;
+    }
+
+    modifier onlyGuardian() {
+        require(isGuardian[msg.sender], "Caller is not a guardian");
+        _;
+    }
+
+    function initialize(string memory _name, string memory _symbol)
+        external
+        initializer
+    {
+        // OpenZeppelin upgradeable contracts documentation says:
+        //
+        // "Use with multiple inheritance requires special care. Initializer
+        // functions are not linearized by the compiler like constructors.
+        // Because of this, each __{ContractName}_init function embeds the
+        // linearized calls to all parent initializers. As a consequence,
+        // calling two of these init functions can potentially initialize the
+        // same contract twice."
+        //
+        // Note that ERC20 extensions do not linearize calls to ERC20Upgradeable
+        // initializer so we call all extension initializers individually. At
+        // the same time, ERC20PermitUpgradeable does linearize the call to
+        // EIP712Upgradeable so we are not using the unchained initializer
+        // versions.
+        __ERC20_init(_name, _symbol);
+        __ERC20Burnable_init();
+        __ERC20Permit_init(_name);
+        __Ownable_init();
+        __Pausable_init();
+    }
+
+    /// @notice Adds the address to the minters list.
+    /// @dev Requirements:
+    ///      - The caller must be the contract owner.
+    ///      - `minter` must not be a minter address already.
+    function addMinter(address minter) external onlyOwner {
```

Comment:
At the end of the day, who will be the `owner`? In RFC it says:
> The Governance will be able to add and remove minters to L2/sidechain tBTC canonical contract.

Don't we need an `onlyGovernance` modifier here instead of `onlyOwner`? The same comment for other `add*`, `remove*` functions.